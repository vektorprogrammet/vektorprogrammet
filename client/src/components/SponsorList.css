ul {
    margin-left: 0;
    padding-left: 0;
}
