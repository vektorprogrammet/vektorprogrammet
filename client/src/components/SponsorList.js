import React, { Component } from 'react';
import './SponsorList.css';

export default class SponsorList extends Component {
  render() {
    const sponsors = this.props.sponsors || [];
    const sponsorItems = sponsors.map(sponsor => (
        <li key={sponsor.id}><a href={sponsor.url}>{sponsor.name}</a></li>
    ));

    return (
        <ul>{sponsorItems}</ul>
    );
  }
}
