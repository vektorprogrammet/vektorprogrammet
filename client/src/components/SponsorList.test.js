import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SponsorList from './SponsorList';

const sponsors = [
  {
    id: 1,
    name: 'Sponsor 1',
    url: 'http://sponsor1.com',
    size: 'small',
    logo_image_path: 'http://placehold.it/100x100',
  },
  {
    id: 2,
    name: 'Sponsor 2',
    url: 'http://sponsor2.com',
    size: 'medium',
    logo_image_path: 'http://placehold.it/10x10',
  },
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SponsorList sponsors={sponsors} />, div);
});

it('renders "undefined" sponsors without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SponsorList />, div);
});

it('renders correctly', () => {
  const tree = renderer.create(
      <SponsorList sponsors={sponsors} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
