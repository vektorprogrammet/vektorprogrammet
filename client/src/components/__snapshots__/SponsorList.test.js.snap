// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <li>
    <a
      href="http://sponsor1.com"
    >
      Sponsor 1
    </a>
  </li>
  <li>
    <a
      href="http://sponsor2.com"
    >
      Sponsor 2
    </a>
  </li>
</ul>
`;
